src/search/search-index/suggestDomains.js

```diff
@@ -0,0 +1,29 @@
+import index from './'
+import { keyGen, removeKeyType } from './util'
+
+/**
+ * @param {string} [query=''] Plaintext query string to match against start of domain names.
+ *  eg. 'wo' would match 'worldbrain.io', 'worldbrain.slack.com' (assuming both these domains exist).
+ * @param {number} [limit=10] Max number of suggestions to return.
+ * @returns {Promise<string[]>} Resolves to an array of matching tags, if any, of length 0 - `limit`.
+ */
+const suggestDomains = (query = '', limit = 10) =>
```

Comment:
Great stuff getting this implemented! :) as it shares a lot of code with the tag suggestion module, maybe it would be better to write a generic "suggest" method for reuse between these indexes? Just find out the things that differ between them and write another function around it (higher order fn) passing those variable things in as args. Not an urgent thing though